### ISSUE TITLE
RFI:  Changing periodicity?

### ISSUE THREAD

**Issue by packet-rat** [adibendahan/mysqlbeat#8]
Total Neophyte with Go.   Need to change the periodicity of mysql queries such that the contents of the mysql table are pulled once every 24 hours (e.g. SELECT * from table WHERE etl_date is...).  

It seems that the delay from period parameter is counted down prior to the first iteration?  So when setting period to 86,400 it is waiting for 24 hours before executing the first iteration.  Is this correct?  How would you change the code to execute the first query prior to starting a delay loop?

Thanks in advance!

**Comment by adibendahan** [comment]
On [beater/mysqlbeat.go](https://github.com/adibendahan/mysqlbeat/blob/master/beater/mysqlbeat.go) change the Run function from:

```
func (bt *Mysqlbeat) Run(b *beat.Beat) error {
	logp.Info("mysqlbeat is running! Hit CTRL-C to stop it.")

	ticker := time.NewTicker(bt.period)
	for {
		select {
		case <-bt.done:
			return nil
		case <-ticker.C:
		}

		err := bt.beat(b)
		if err != nil {
			return err
		}
	}
}
```

To:

```
func (bt *Mysqlbeat) Run(b *beat.Beat) error {
	logp.Info("mysqlbeat is running! Hit CTRL-C to stop it.")

	err := bt.beat(b)
	if err != nil {
		return err
	}

	ticker := time.NewTicker(bt.period)
	for {
		select {
		case <-bt.done:
			return nil
		case <-ticker.C:
		}

		err := bt.beat(b)
		if err != nil {
			return err
		}
	}
}
```

That way it'll run first and start a timer to continue running next iteration...

-- Adi.

**Comment by packet-rat** [comment]
Many thanks!!! - I figured this out myself.  In our case (since we want only want to run this once a day against a dozen tables) I removed the entire ticker section and added your first section above.  I also needed a "return nil" to close out the block.

This allows us to run a script with each mysql table==>es index update in series.  We were seeing spurious issues when trying to run these concurrently against our es cluster.

...and many thanks for these useful scripts!!!
